http-protocol-binding.md

```diff
@@ -213,16 +213,75 @@ and per [RFC7230, section 3][rfc7230-section-3], HTTP headers MUST only use
 printable characters from the US-ASCII character set, and are terminated by a
 CRLF sequence with OPTIONAL whitespace around the header value.
 
-String values MUST be percent-encoded as described in [RFC3986, section
-2.4][rfc3986-section-2-4] before applying the header encoding rules described in
-[RFC7230, section 3.2.6][rfc7230-section-3-2-6].
-
-When decoding an HTTP message into a CloudEvent, these rules MUST be applied in
-reverse -- [RFC7230, section 3.2.6][rfc7230-section-3-2-6] decoding to an ASCII
-string, and then a **single round** of percent-decoding as described in
-[RFC3986, section 2.4][rfc3986-section-2-4] to produce a valid UTF-8 String.
-(Note that applying percent-decoding an incorrect number of times can result in
-message corruption or security issues.)
+When encoding a CloudEvent as an HTTP message, string values
+represented as HTTP header values MUST be percent-encoded as
+described below. This is compatible with [RFC3986, section
+2.1][rfc3986-section-2-1] but is more specific about what needs
+encoding. The resulting string does not then need any further
+quoting, due to space and quote characters being encoded.
+
+When decoding an HTTP message into a CloudEvent, any quoted string
+as described in [RFC7230, section 3.2.6][rfc7230-section-3-2-6]
+appearing as an HTTP header value MUST be decoded to an ASCII
```

Comment:
The quoted string is already US-ASCII per definition and with the kinda-sorta exceptions discussed in RFC7230, 3.2.4 in the last paragraph. I would just omit "decoded to an ASCII string" here because that's what it ought to be already and it's confusing what the decoding action ought to be.

"...appearing as an HTTP header value (limited to US-ASCII per RTC730) must be percent-decoded as described below" 

maybe?

Reply:
Right. I think this was a matter of keeping too much of what was already there.

But we do need to specify that "unquoting" is performed. How about this:

> When decoding an HTTP message into a CloudEvent, any HTTP header
> value MUST have quoted string escaping as described in
> [RFC7230, section 3.2.6][rfc7230-section-3-2-6] applied in reverse, and
> then a **single round** of percent-decoding MUST be performed as
> described below.

(Remainder of paragraph as-is.)

Reply:
Better.